### PR TITLE
Create an Environment Class 

### LINKED ISSUE #30: Create an Environment class
The Environment class will encompass parameters that define an air parcel,
such as temperature, pressure, etc. and any derived variables.
The goal is to allow changes in air parcels, such as dilution/compression, 
warming/cooling, and pressure changes to occur through the change of a set 
of parameters, instead of individual changes to environmental constants.

### PR DESCRIPTION
**Pull request by wkchuang**
Creates an Environment class to set the properties of an air parcel, such as temperature and pressure.

Fixes #30

### CODE REVIEW

**Comment by ngam** [comment]
@wkchuang if these linting errors are bothersome, I can either go in and make the linters happy or we can relax the conditions for failure with pylint (currently, the ci isn't going through because of pylint --- it's not yet checking your actual code; that's why I want to split them, see #38)

**Comment by ngam** [comment]
The slip correction testing is failing...

**Comment by ngam** [comment]
I fixed the tests by adding decimals to the pytest calls. We should move away from these numerical tests for now and focus on more general tests. I will think more about this and do another PR to unify our testing schema

**Comment by ngam** [comment]
@wkchuang, diff between your commit and my last commit is below (without the requested/suggested edits):

```diff

diff --git a/aerosol_dynamics/environment.py b/aerosol_dynamics/environment.py
index a3be357..6588c76 100644
--- a/aerosol_dynamics/environment.py
+++ b/aerosol_dynamics/environment.py
@@ -1,6 +1,12 @@
+"""
+Env Class
+"""
+
+import math
+
 from aerosol_dynamics import physical_parameters as pp
 from . import u
-import math
+
 
 class Environment:
     """
@@ -26,34 +32,34 @@ class Environment:
     @u.wraps(u.kg / u.m / u.s, [None])
     def dynamic_viscosity_air(self) -> float:
         """Returns the dynamic viscosity of air. [kg/m/s]
-        
+
         The dynamic viscosity is calculated using the 3 parameter
         Sutherland Viscosity Law.
         """
-        MU_REF = 1.716e-5 * u.Pa * u.s # Viscosity at T_REF
-        T_REF = 273.15 * u.K
-        S = 110.4 * u.K # Sutherland constant
+        mu_ref = 1.716e-5 * u.Pa * u.s # Viscosity at T_REF
+        t_ref = 273.15 * u.K
+        suth_const = 110.4 * u.K # Sutherland constant
 
         return (
-            MU_REF *
-            (self.temperature()/T_REF)**(3/2) *
-            (T_REF+S) / (self.temperature()+S)
+            mu_ref *
+            (self.temperature()/t_ref)**(3/2) *
+            (t_ref + suth_const) / (self.temperature() + suth_const)
         )
 
     # mean free path of air in m
     # @u.wraps(u.m, [None])
     def mean_free_path_air(self) -> float:
         """Returns the mean free path of this environment. [m]
-        
+
         The mean free path is the average distance traveled by a molecule
         between collisions with other molecules.
         """
-        MOLECULAR_WEIGHT = (28.9644 * u.g / u.mol).to_base_units() # Molecular weight of air
-        
+        molecular_weight = (28.9644 * u.g / u.mol).to_base_units() # Molecular weight of air
+
         return (
             (
                 2 * self.dynamic_viscosity_air() /
                 self.pressure() /
-                (8*MOLECULAR_WEIGHT / (math.pi*pp.GAS_CONSTANT*self.temperature()))**0.5
+                (8*molecular_weight / (math.pi*pp.GAS_CONSTANT*self.temperature()))**0.5
             ).to_base_units()
         )
diff --git a/aerosol_dynamics/environment_test.py b/aerosol_dynamics/environment_test.py
index bfe6114..943c74e 100644
--- a/aerosol_dynamics/environment_test.py
+++ b/aerosol_dynamics/environment_test.py
@@ -1,22 +1,35 @@
-from aerosol_dynamics import environment
+"""
+testing new env class
+"""
+
 import pytest
+
+from aerosol_dynamics import environment
+
 from . import u
 
+
 standard_environment = environment.Environment(
     temperature=298 * u.K,
     pressure=101325 * u.Pa,
 )
 
 def test_getters():
+    """testing getters
+    """
     assert standard_environment.temperature() == 298 * u.K
     assert standard_environment.pressure() == 101325 * u.Pa
 
 def test_dynamic_viscosity_air():
+    """testing dynamic visc
+    """
     assert (standard_environment.dynamic_viscosity_air().magnitude ==
         pytest.approx(1.836e-5, rel=1e-3)
         )
 
 def test_mean_free_path_air():
+    """testing mfp
+    """
     assert (standard_environment.mean_free_path_air().magnitude ==
         pytest.approx(6.644e-8, rel=1e-3)
-        )
\ No newline at end of file
+        )
diff --git a/aerosol_dynamics/particle.py b/aerosol_dynamics/particle.py
index 5306cc3..6e4f4c4 100644
--- a/aerosol_dynamics/particle.py
+++ b/aerosol_dynamics/particle.py
@@ -145,17 +145,23 @@ class Particle:
         """
 
         reduced_mass = self.reduced_mass(other)
-        coulomb_enhancement_continuum_limit = self.coulomb_enhancement_continuum_limit(other, environment)
+        coulomb_enhancement_continuum_limit = self.coulomb_enhancement_continuum_limit(
+            other, environment
+        )
         reduced_friction_factor = self.reduced_friction_factor(other, environment)
-        coulomb_enhancement_kinetic_limit = self.coulomb_enhancement_kinetic_limit(other, environment)
+        coulomb_enhancement_kinetic_limit = self.coulomb_enhancement_kinetic_limit(
+            other, environment
+        )
         return (
-            (environment.temperature() * pp.BOLTZMANN_CONSTANT * reduced_mass**0.5) 
+            (environment.temperature() * pp.BOLTZMANN_CONSTANT * reduced_mass**0.5)
             * coulomb_enhancement_continuum_limit /
             (reduced_friction_factor * (self.radius() + other.radius())
             * coulomb_enhancement_kinetic_limit)
         )
 
-    def dimensionless_coagulation_kernel_hard_sphere(self, other, environment: Environment) -> float:
+    def dimensionless_coagulation_kernel_hard_sphere(
+        self, other, environment: Environment
+    ) -> float:
         """
         Calculates the dimensionless coagulation kernel for a particle-particle interaction
         """
diff --git a/aerosol_dynamics/particle_test.py b/aerosol_dynamics/particle_test.py
index 6bb0770..ee19226 100644
--- a/aerosol_dynamics/particle_test.py
+++ b/aerosol_dynamics/particle_test.py
@@ -56,7 +56,7 @@ def test_knudsen_number():
     """
     Test that the knudsen number is calculated correctly.
     """
-    
+
     assert small_particle.knudsen_number(standard_environment) == pytest.approx(66.4374)
     assert small_particle.knudsen_number(standard_environment).check(["None"])
     assert large_particle.knudsen_number(standard_environment) == pytest.approx(0.664374)
@@ -73,8 +73,8 @@ def test_slip_correction_factor():
     Test that the slip correction factor is calculated correctly.
     """
 
-    assert small_particle.slip_correction_factor(standard_environment) == pytest.approx(110.650)
-    assert large_particle.slip_correction_factor(standard_environment) == pytest.approx(1.88587)
+    assert small_particle.slip_correction_factor(standard_environment) == pytest.approx(110.6503)
+    assert large_particle.slip_correction_factor(standard_environment) == pytest.approx(1.885865)
     assert small_particle.slip_correction_factor(standard_environment).check(["None"])
 
 
@@ -83,9 +83,15 @@ def test_friction_factor():
     Test that the friction factor is calculated correctly.
     """
 
-    assert small_particle.friction_factor(standard_environment).magnitude == pytest.approx(3.18080e-15)
-    assert large_particle.friction_factor(standard_environment).magnitude == pytest.approx(1.84e-11)
-    assert small_particle.friction_factor(standard_environment).check("[mass]/[time]")
+    assert small_particle.friction_factor(
+        standard_environment
+    ).magnitude == pytest.approx(3.18080e-15)
+    assert large_particle.friction_factor(
+        standard_environment
+    ).magnitude == pytest.approx(1.84e-11)
+    assert small_particle.friction_factor(
+        standard_environment
+    ).check("[mass]/[time]")
 
 
 def test_reduced_mass():

```

File: aerosol_dynamics/particle_test.py [hunk 1]
```diff
@@ -31,12 +31,15 @@
     charge=3 * u.C,
```
[the remark this answers is not in the document]
Comment (wkchuang) [reply aerosol_dynamics/particle_test.py:31]: That's the point of this particle. It's invalid and intended to be either unrealistic or with the wrong units.

File: aerosol_dynamics/particle_test.py [hunk 2]
```diff
@@ -71,27 +71,27 @@ def test_knudsen_number():
...
     Test that the slip correction factor is calculated correctly.
-    TODO:
-    - can we check units instead of numbers?
-    - that's an overly specific number with lots of digs
     """
 
-    assert small_particle.slip_correction_factor() == pytest.approx(108.268702)
-    assert large_particle.slip_correction_factor() == pytest.approx(1.864914)
-    assert small_particle.slip_correction_factor().check(["None"])
+    assert small_particle.slip_correction_factor(standard_environment) == pytest.approx(110.650)
```
[the remark this answers is not in the document]
Comment (wkchuang) [reply aerosol_dynamics/particle_test.py]: I've made changes that allow it to pass now with fewer digits by defining a relative tolerance of 1e-3.

File: aerosol_dynamics/environment.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+"""
+Env Class
+"""
+
+import math
```
[the remark this answers is not in the document]
Comment (wkchuang) [reply aerosol_dynamics/environment.py:5]: good idea, more consistent package usage

File: aerosol_dynamics/environment.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        suth_const = 110.4 * u.K # Sutherland constant
+
+        return (
+            mu_ref *
+            (self.temperature()/t_ref)**(3/2) *
+            (t_ref + suth_const) / (self.temperature() + suth_const)
+        )
+
+    # mean free path of air in m
+    # @u.wraps(u.m, [None])
```
[the remark this answers is not in the document]
Comment (wkchuang) [reply aerosol_dynamics/environment.py:50]: That was commented out while testing, thanks for catching it. It should be kept.

**Comment by wkchuang** [comment]
This is a small thing and I won't press this, but I dislike the blockiness of test_friction_factor when everything's vertical. It's harder to vertically compare them, especially when the code has a lot of the same words with tiny changes to cover different cases.

**Comment by ngam** [comment]
> This is a small thing and I won't press this, but I dislike the blockiness of test_friction_factor when everything's vertical. It's harder to vertically compare them, especially when the code has a lot of the same words with tiny changes to cover different cases.

Well... pylint will fail your run then... see what happened just now. We can override it, or we can work around it.

**Review by ngam — APPROVED** [review]
Looks good!

**Pull request merged**